relay-server/src/endpoints/common.rs

```diff
@@ -344,85 +299,65 @@ where
     let project_key = meta.public_key();
     let start_time = meta.start_time();
     let config = request.state().config();
-
-    let envelope_context = Rc::new(RefCell::new(EnvelopeContext::from_request(&meta)));
+    let event_id = Rc::new(RefCell::new(None));
```

Comment:
The `EnvelopeContext` is now moved through the chain of futures rather than shared across all the callbacks. However, we still need the EventId in the global error handler, so we can't get rid of the `RefCell` just yet.